test/e2e/gateway_api_test.go

```diff
@@ -265,11 +272,11 @@ func testGatewayAPIResourcesProtection(t *testing.T) {
 func ensureCRDs(t *testing.T) {
 	t.Helper()
 	for _, crdName := range crdNames {
-		crdVersion, err := assertCrdExists(t, crdName)
+		crdVersions, err := assertCRDExists(t, crdName)
 		if err != nil {
 			t.Fatalf("failed to find crd %s: %v", crdName, err)
 		}
-		t.Logf("found crd %s at version %s", crdName, crdVersion)
+		t.Logf("Found CRD %s with the following served versions: %s", crdName, strings.Join(crdVersions, ", "))
```

Comment:
Do you think it might be better to change `assertCRDExists` to make it check for served and storage and ignore the CRD if storage is true?

Also, nit:
```suggestion
		t.Logf("found CRD %s with the following served versions: %s", crdName, strings.Join(crdVersions, ", "))
```

Reply:
Log messages should start with a capital letter by Kubernetes conventions.  Source: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/logging.md#message-style-guidelines